File: algorithms/1588.sum-of-all-odd-length-subarrays/solution.rb
# @param {Integer[]} arr
# @return {Integer}
def sum_odd_length_subarrays(arr)
  count = 0
  i = 0
  while i < arr.size
    j = i
    while j < arr.size
      count += arr[i..j].sum
      j += 2
    end
    i += 1
  end

  count
end

File: algorithms/1588.sum-of-all-odd-length-subarrays/solution.go
func sumOddLengthSubarrays(arr []int) int {
	count := 0

	for i := 1; i <= len(arr); i += 2 {
		for j := 0; j < len(arr)-i+1; j++ {
			for k := j; k < i+j; k++ {
				count += arr[k]
			}
		}
	}
	return count
}
